packages/manager/apps/dedicated/client/app/dedicated/server/installation/ovh/dedicated-server-installation-ovh.controller.js

```diff
@@ -2017,11 +2017,41 @@ angular
       $scope.$watch('installation.hardwareRaid.raid', () => {
         $scope.clearHardwareRaidSpace();
         $scope.recalculateAvailableRaidDisks();
+        if ($scope.informations.hardwareRaid.availableDisks.length === 1) {
+          [
+            $scope.installation.hardwareRaid.disks,
+          ] = $scope.informations.hardwareRaid.availableDisks;
+          $scope.recalculateAvailableArrays();
+        }
+        if ($scope.informations.hardwareRaid.availableArrays.length === 1) {
+          [
+            $scope.installation.hardwareRaid.arrays,
+          ] = $scope.informations.hardwareRaid.availableArrays;
+          $scope.recalculateSpace();
+          if (
+            $scope.installation.hardwareRaid.disks &&
+            $scope.installation.hardwareRaid.arrays
+          ) {
+            $scope.installation.hardwareRaid.error = $scope.invalidHardRaid();
+          }
```

Comment:
better for visibility

```suggestion
        const {availableDisks, availableArrays, disks, arrays } = $scope.informations.hardwareRaid;
        if (availableDisks.length === 1) {
          [ hardwareRaid.disks ] = availableDisks;
          $scope.recalculateAvailableArrays();
        }
        if (availableArrays.length === 1) {
          [ hardwareRaid.arrays ] = availableArrays;
          $scope.recalculateSpace();
          if ( disks && arrays) {
            $scope.installation.hardwareRaid.error = $scope.invalidHardRaid();
          }
```